### ISSUE TITLE
"Non-native" Fullscreen on OSX?

### ISSUE THREAD

**Issue by bnferguson** [atom/atom#967]
This is not a huge priority and may even be a "wontfix" (which is completely ok)

Currently when using Fullscreen in Atom it does what nearly every Mac app does post 10.7, creates a new space, and goes full screen over there. 

This is a bummer since it kills your second screen, and any cmd-tabs will cause screens to flip back and forth. Also can't have a browser, Chat, or what have you up side by side with it, etc.

MacVim has this as an option (which I keep off): 
![advanced 2013-10-14 16-13-08](https://f.cloud.github.com/assets/47749/1328928/0a8f68ea-350d-11e3-93ba-64df6da09eeb.jpg)

Is it possible to have this in Atom or even just something like it in my ~/.atom/user.coffee?

Totally recognize that this is a little finicky. :grin:

**Comment by evantravers** [comment]
:+1: Please!

**Comment by seanhealy** [comment]
+1

Been wanting this in Atom since I stated using it.

**Comment by drslump** [comment]
FWIW with [hyperswitch](http://bahoom.com/hyperswitch) (free) it's possible to have a keyboard shortcut to cycle through different windows in an App even if they are in different *spaces* (unlike the native <kbd>Cmd+`</kbd>). Furthermore, with [TotalSpaces2](http://totalspaces.binaryage.com/) (pay) *space* switching animations can be completely disabled.

That has solved most of my troubles with the native fullscreen mode from OSX since I don't use a secondary monitor, although it would indeed be great if Atom provided a fake-fullscreen mode like iTerm2 or MacVim.

**Comment by forrest** [comment]
+1

**Comment by cyberwombat** [comment]
+1

**Comment by robink** [comment]
+1

**Comment by gokmen** [comment]
+1

**Comment by caiogondim** [comment]
Would love to see it implemented

**Comment by maletor** [comment]
This is a very important part of productivity.

**Comment by philipgiuliani** [comment]
Try to double-click on the window title bar :) Works on OS X Yosemite in every app. Im not sure how its in other OSX Versions!

**Comment by takashi** [comment]
+1

**Comment by maletor** [comment]
Has anybody looked at how macvim implements it and if it can be duplicated?

**Comment by cperryk** [comment]
+1

**Comment by jamesperet** [comment]
+1

**Comment by specik** [comment]
+1

**Comment by murphyrandle** [comment]
+1

**Comment by fullofcaffeine** [comment]
+1

**Comment by davedan7** [comment]
the lack of this feature is the main thing keeping me from switching to Atom from ST3

**Comment by landongn** [comment]
another +1 for the pile. This is something that I use constantly in ST3.

**Comment by AlexKvazos** [comment]
+1
Only thing I need to have 0 reasons to go back to sublime

**Comment by boeserwolf** [comment]
+1
> the lack of this feature is the main thing keeping me from switching to Atom from ST3

**Comment by ellipticaldoor** [comment]
+1

**Comment by sugy** [comment]
+1

**Comment by fabianmarz** [comment]
+1

**Comment by mnquintana** [comment]
Thanks for the feedback everyone! However, all these +1s and :+1:s are just adding noise, so to minimize that I'm going to lock this issue for now – we may unlock it once someone starts working on this enhancement to get further feedback.

**Comment by 50Wliu** [comment]
Update: this has been added in https://github.com/electron/electron/pull/10254 and will probably be released in Electron 1.8.x.  I'll be adding this issue to the list of things to consider when we begin upgrading to a version of Electron that contains electron/electron#10254.

**Comment by jasonrudolph** [comment]
> I'll be adding this issue to the list of things to consider when we begin upgrading to a version of Electron that contains electron/electron#10254.

> In other words, we'll have to upgrade to Electron v1.8.2 or newer.

Linking these Electron upgrade pull requests:

- https://github.com/atom/atom/pull/16755 - Electron 1.8
- https://github.com/atom/atom/pull/17273 - Electron 2.0

Once one of those pull requests is merged, this issue should become unblocked.

**Comment by lkashef** [comment]
fixed by https://github.com/atom/atom/pull/20764

**Comment by sadick254** [comment]
We now have a PR for adding non-native fullscreen to atom. https://github.com/atom/atom/pull/21214
